playground/src/components/playground/PlaygroundInput.tsx

```diff
@@ -22,35 +32,79 @@ export function PlaygroundInput({
       //        That's why we add `m-0` to the form element.
-      className="flex items-center justify-between bg-input border rounded-lg m-0"
+      className="flex items-center justify-between bg-input border rounded-lg mx-0 py-1"
     >
-      <div className="flex flex-grow items-center space-x-0">
-        <RequestMethodCombobox
-          method={method}
-          handleMethodChange={handleMethodChange}
-          allowUserToChange
-        />
-        <Input
-          type="text"
-          value={path}
-          onChange={(e) => {
-            handlePathInputChange(e.target.value);
-          }}
-          className="flex-grow text-xs w-full bg-transparent font-mono border-none shadow-none focus:ring-0 ml-0"
-        />
+      <div className="flex-1 min-w-0">
+        <FpDropdownMenu>
+          <FpDropdownMenuTrigger
+            className={cn(
+              "flex",
+              "ml-2",
+              "items-center",
+              "gap-2",
+              "h-min",
+              "hover:bg-muted",
+              "data-[state=open]:bg-muted",
+              "rounded-sm",
+              "w-full",
+            )}
+          >
+            <Method
+              method={activeRoute?.method || "GET"}
+              className="ml-2 text-sm flex-shrink-0"
+            />
+            <div className="flex-1 justify-start min-w-0">
+              <div className="text-xs font-mono truncate text-left">{path}</div>
+            </div>
+            <CaretSortIcon className="w-3 h-3 mr-1 flex-shrink-0" />
+          </FpDropdownMenuTrigger>
+          <FpDropdownMenuPortal>
+            <FpDropdownMenuContent align="start">
+              <FpDropdownMenuRadioGroup
+                value={activeRoute ? getRouteId(activeRoute) : ""}
+              >
+                {appRoutes.map((route) => (
+                  <FpMinimalDropdownMenuRadioItem
+                    key={getRouteId(route)}
+                    onSelect={() => setActiveRoute(route)}
+                    value={getRouteId(route)}
+                    className={cn(
+                      "py-0.25",
+                      "focus:bg-muted",
+                      "aria-checked:bg-muted aria-checked:focus:text-accent-foreground",
+                    )}
+                  >
+                    <div
+                      className={cn(
+                        "flex-grow grid grid-cols-[2.5rem_auto] items-center gap-1 px-1 rounded-md",
+                      )}
+                    >
+                      <Method method={route.method} className="ml-1 text-xs" />
+                      <div className="flex w-full my-2">
+                        <div className="flex-initial text-xs w-full bg-transparent font-mono border-none shadow-none focus:ring-0 ml-0 disabled:cursor-text disabled:bg-muted py-0">
+                          {route.summary || route.description || route.path}
```

Comment:
i removed `ApiRoute["title"]` as OpenAPI only specifies "summary" and "description" fields 

for our purposes "summary" is the route title

wondering if we also wanna show description in this dropdown if it's available?

<img width="401" alt="image" src="https://github.com/user-attachments/assets/14f8a156-1aab-4345-8ac5-b398df927c3a" />

Reply:
Perhaps a designer can make a nice suggestion here. This dropdown is a bit clunky